### PR TITLE
docs(devtools): Documentation on how to use DevtoolsLogger for Fluid Devtools

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Add documentation for developers building Fluid-powered apps on how to use the `DevtoolsLogger`, a necessary piece to enable the full functionality of Fluid Devtools.

The devtools README has a reference to `AzureClient` and the devtools-core one to `Loader`, and each one points to the other one for developers using `AzureClient` VS `Loader`.

## Reviewer Guidance

Looking for feedback on the docs themselves (clarity, completeness) and any thoughts if there's a better place for them. They should eventually be in fluidframework.com, but while we sort that out this seems like the most natural place for them.

[AB#3633](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3633)

### CODE REVIEW

File: packages/tools/devtools/devtools-core/README.md [hunk 1]
```diff
@@ -29,10 +29,56 @@ The Devtools' API surface is designed to fit nicely into most application flows.
 ### Initialization
 
 To initialize a debugger session for your container, call `initializeDevtools`.
-This function accepts a `DevtoolsLogger` for recording and communicating telemetry data, a list of initial Fluid `Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing `Container` data.
+This function accepts a `DevtoolsLogger` for receiving Fluid telemetry data from your application, a list of initial Fluid
+`Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing
+`Container` data.
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-core/README.md:34]: Just breaking the long line into several.

File: packages/tools/devtools/devtools/README.md [hunk 2]
```diff
@@ -26,10 +26,60 @@ The Devtools' API surface is designed to fit nicely into most application flows.
 ### Initialization
 
 To initialize a debugger session for your container, call `initializeDevtools`.
-This function accepts a `DevtoolsLogger` for recording and communicating telemetry data, a list of initial Fluid `Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing `Container` data.
+This function accepts a `DevtoolsLogger` for recording and communicating telemetry data, a list of initial Fluid
+`Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing
+`Container` data.
```
Comment (alexvy86) [inline packages/tools/devtools/devtools/README.md:31]: Just breaking the long line into several.

File: packages/tools/devtools/devtools-core/README.md [hunk 3]
```diff
@@ -29,10 +29,56 @@ The Devtools' API surface is designed to fit nicely into most application flows.
 ### Initialization
 
 To initialize a devtools session for your container, call `initializeDevtools`.
-This function accepts a `DevtoolsLogger` for recording and communicating telemetry data, a list of initial Fluid `Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing `Container` data.
+This function accepts a `DevtoolsLogger` for receiving Fluid telemetry from your application, a list of initial Fluid
+`Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing
+`Container` data.
 
 TODO: link to API docs once API shape has settled.
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-core/README.md:36]: I thought I had added this recently. Could we link to the docs here? https://fluidframework.com/docs/apis/devtools-core#initializedevtools-function

File: packages/tools/devtools/devtools/README.md [hunk 4]
```diff
@@ -26,10 +26,60 @@ The Devtools' API surface is designed to fit nicely into most application flows.
 ### Initialization
 
 To initialize a devtools session for your container, call `initializeDevtools`.
-This function accepts a `DevtoolsLogger` for recording and communicating telemetry data, a list of initial Fluid `Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing `Container` data.
+This function accepts a `DevtoolsLogger` for receiving Fluid telemetry from your application, a list of initial Fluid
+`Containers` to associate with the session, and (optionally) customized data visualization configurations for visualizing
+`Container` data.
 
 TODO: link to API docs once API shape has settled.
```
Comment (Josmithr) [inline packages/tools/devtools/devtools/README.md:33]: Can we link to the docs here? https://fluidframework.com/docs/apis/devtools#initializedevtools-function

File: packages/tools/devtools/devtools/README.md [hunk 5]
```diff
@@ -26,10 +26,60 @@ The Devtools' API surface is designed to fit nicely into most application flows.
...
 
+#### About the `DevtoolsLogger`
```
Comment (Josmithr) [inline packages/tools/devtools/devtools/README.md:35]: In case we want to link to the API docs for the class, there are here: https://fluidframework.com/docs/apis/devtools-core/devtoolslogger-class

File: packages/tools/devtools/devtools-core/README.md [hunk 6]
```diff
@@ -29,10 +29,56 @@ The Devtools' API surface is designed to fit nicely into most application flows.
...
+const devtools = initializeDevtools({ logger: devtoolsLogger });
+
+const loader = new Loader({
+    logger: devtoolsLogger,
+    // Other necessary properties ...
+});
+```
+
+If you're working with `AzureClient` instead of lower-level APIs like the one described above, you probably want to
+refer to the [@fluidframework/devtools package](../devtools/) instead of this one.
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-core/README.md:62]: Relative file paths like these will work fine in the context of the repo, but these READMEs also get published alongside the packages to npm, where the relative links won't work. We should probably use explicit github links instead.

Comment (Josmithr) [reply packages/tools/devtools/devtools-core/README.md:62]: Either that or link to the relevant npm page. That page will have a link back to the github anyways.

**Review by Josmithr — COMMENTED** [review]
I left a few nitpicky recommendations, otherwise my only major feedback is regarding our use of relative file paths for package links. I think we need to link to npm / github via explicit URLs instead so that the links work on the npmjs pages for these packages.

The contents look great to me! Thanks for updating the docs!!

File: packages/tools/devtools/devtools/README.md [hunk 7]
```diff
@@ -26,10 +26,60 @@ The Devtools' API surface is designed to fit nicely into most application flows.
...
+This way the logger will receive all the telemetry generated by Fluid Framework in your application, and forward it to
+Fluid Devtools as necessary.
+For example, when using the `AzureClient` API you would do something like this (note how `devtoolsLogger` is passed
+to `initializeDevtools()` _and_ to `new AzureClient()`):
+
+```typescript
+import { DevtoolsLogger, initializeDevtools } from "@fluid-experimental/devtools-core";
+
+const devtoolsLogger = new DevtoolsLogger();
+const devtools = initializeDevtools({ logger: devtoolsLogger });
```
[the remark this answers is not in the document]
Comment (alexvy86) [reply packages/tools/devtools/devtools/README.md:52]: Good call, I'll move it and add the container to the props.

File: packages/tools/devtools/devtools-core/README.md (hunk 6 above)
Comment (alexvy86) [reply packages/tools/devtools/devtools-core/README.md:62]: I was a bit torn which of the two to use. I liked the github links, but they're more prone to breaking if we move things around. The npm ones are more stable, but require another click (or more) to go back to the code for the package; which might not be the main use case of the links in the README so that's probably ok. And those will need to be updated when we move from the `@fluid-experimental` scope. But it felt like the way to go.

File: packages/tools/devtools/devtools/README.md [hunk 8]
```diff
@@ -29,10 +29,75 @@ The Devtools' API surface is designed to fit nicely into most application flows.
...
-See [here](https://fluidframework.com/docs/apis/devtools#initializedevtools-function) for associated API documentation.
+// Your application's logger
+const yourApplicationLogger = getInstanceOfYourApplicationLogger();
+
+const devtoolsLogger = new DevtoolsLogger(yourApplicationLogger);
+
+// Pass devtoolsLogger to initializeDevtools() and to your application as described above
+```
 
 ### Clean-up
```
Comment (ghoshkaj) [inline packages/tools/devtools/devtools/README.md:103]: All your changes are good. But I do have a suggestion for the "Working in the package" title below. I find "Working in the package" to be confusing - I think it's referring to contributing to changes in the package right? Maybe rename it to "Contributing to the package"? Could do in a different PR too since that change is unrelated to your additions above!

Comment (alexvy86) [reply packages/tools/devtools/devtools/README.md:103]: Yeah, that's what that section is about. We can make that change in a separate PR as you suggest.

Comment (alexvy86) [reply packages/tools/devtools/devtools/README.md:103]: Had to make another fix so I went ahead and updated the header too 👍

**Review by seanimam — APPROVED** [review]
